File: android/app/src/main/java/ru/smarthome/database/store/RPInfoStorage.java
package ru.smarthome.database.store;

import com.google.firebase.database.DatabaseReference;

import ru.smarthome.database.store.listeners.RPInfoListener;

public interface RPInfoStorage {

    void postRaspberryIp(String ip);

    void postRaspberryIp(String ip, DatabaseReference.CompletionListener listener);

    void getRaspberryInfo(RPInfoListener listener);

}

File: android_client/android/app/src/main/java/ru/smarthome/database/store/RPInfoStorage.kt
package ru.smarthome.database.store

import com.google.firebase.database.DatabaseReference

import ru.smarthome.database.store.listeners.RPInfoListener

interface RPInfoStorage {
    fun postRaspberryIp(ip: String)
    fun postRaspberryIp(ip: String, listener: DatabaseReference.CompletionListener)
    fun getRaspberryInfo(listener: RPInfoListener)
}
